### PR TITLE
integrals: fixed bug which produced a wrong result when integrating derivatives

### PR DESCRIPTION
**Pull request by 2torus**
Fixes issue #12899. The original logic was faulty in several ways:
a) treating a derivative with respect to a variable different
from integration symbol as if it was constant.
b) looking only at the last variable of the differentiation.
Fixed a test in test_manual. Added new tests in test_manual
This makes a mild assumption that integration and differentiation commute.

### CODE REVIEW

**Comment by asmeurer** [comment]
Currently in master

```py
>>> diff(Integral(f(x, y), y), x)
Integral(Derivative(f(x, y), x), y)
```

So we already assume that. But I wonder if we should pick one, as otherwise it would just alternate on repeated calls to `doit`. Or maybe that's a good thing.

File: sympy/integrals/manualintegrate.py [hunk 1]
```diff
@@ -1153,7 +1163,17 @@ def eval_derivativerule(integrand, symbol):
     if len(integrand.args) == 2:
         return integrand.args[0]
     else:
-        return sympy.Derivative(integrand.args[0], *integrand.args[1:-1])
+        # we are removing the last occurence of symbol
+        # from the list of differentiating variables
+        diff_vars = tuple(reversed(integrand.args[1:]))
+        symbol_index = diff_vars.index(symbol)
+        diff_vars_symbol_removed = diff_vars[:symbol_index] + diff_vars[symbol_index + 1:]
```
Comment (asmeurer) [inline sympy/integrals/manualintegrate.py]: You should be able to use `diff_vars.remove(symbol)` (if `diff_vars` is a list).

File: sympy/integrals/manualintegrate.py [hunk 2]
```diff
@@ -1153,7 +1163,17 @@ def eval_derivativerule(integrand, symbol):
     if len(integrand.args) == 2:
         return integrand.args[0]
     else:
-        return sympy.Derivative(integrand.args[0], *integrand.args[1:-1])
+        # we are removing the last occurence of symbol
+        # from the list of differentiating variables
```
Comment (asmeurer) [inline sympy/integrals/manualintegrate.py]: Differentiation variables are sorted. The logic here is not needed.

File: sympy/integrals/manualintegrate.py (hunk 1 above)
Comment (2torus) [inline sympy/integrals/manualintegrate.py]: It's a tuple at the moment but making it a list is more natural here.

File: sympy/integrals/manualintegrate.py (hunk 2 above)
Comment (2torus) [inline sympy/integrals/manualintegrate.py]: Logic removed.

**Comment by jksuom** [comment]
This looks good. Only some lines have grown out of bounds.

**Comment by 2torus** [comment]
@jksuom I'll break up the long lines I introduced.

**Comment by 2torus** [comment]
@asmeurer Let me marinate on your remark regarding exchanging integral and derivatives, I might have been overzealous there. 
I fixed everything else in the last commit.

File: sympy/integrals/tests/test_manual.py [hunk 3]
```diff
@@ -316,5 +317,11 @@ def test_issue_10847():
...
     assert manualintegrate(log(y)/(y**2*(1 - 1/y)), y) == (-log(y) + log(y - 1))*log(y) + log(y)**2/2 - Integral(log(y - 1)/y, y)
 
-def test_constant_independent_of_symbol():
+def test_issue_12899():
+    assert manualintegrate(f(x,y).diff(x),y) == Derivative(Integral(f(x,y),y),x)
+    assert manualintegrate(f(x,y).diff(y).diff(x),y) == Derivative(f(x,y),x)
+
+
+
+def test_constant_independeint_of_symbol():
```
Comment (jksuom) [inline sympy/integrals/tests/test_manual.py]: typo: independeint

**Comment by 2torus** [comment]
@asmeurer @jksuom Could you review the latest changes?

**Pull request merged**